src/plugins/data_view_management/public/components/index_pattern_table/index_pattern_table.tsx

```diff
@@ -88,29 +86,42 @@ export const IndexPatternTable = ({
     docLinks,
   } = useKibana<IndexPatternManagmentContext>().services;
   const [query, setQuery] = useState('');
-  const [indexPatterns, setIndexPatterns] = useState<IndexPatternTableItem[]>([]);
-  const [isLoadingIndexPatterns, setIsLoadingIndexPatterns] = useState<boolean>(true);
   const [showCreateDialog, setShowCreateDialog] = useState<boolean>(showCreateDialogProp);
   const [selectedItems, setSelectedItems] = useState<IndexPatternTableItem[]>([]);
-  const [isLoadingDataState, setIsLoadingDataState] = useState<boolean>(true);
-  const [dataState, setDataState] = useState<DataState>({});
+  const [dataViewController] = useState(
+    () =>
+      new DataViewTableController({
+        services: { dataViews },
+        config: { defaultDataView: uiSettings.get('defaultIndex') },
+      })
+  );
```

Comment:
QQ: having a first look. this is state that never changes, so why not using `useRef` for that? We've got a hook for that making it convenient: https://github.com/elastic/kibana/blob/581d6f8592a2c44d524d5ed3afd4b0be254068eb/src/plugins/discover/public/application/main/hooks/use_singleton.ts

Reply:
Well, you're asking me to use useRef but then you're handing me a shortcut to use it. I guess I don't have a super strong opinion on this, its just whatever gets the job done - but I'd prefer to use something thats the same across kibana.

Reply:
apologies to be unprecise. I just wanted to give a quick example how it's being used in the Discover world. I do agree something like this could be a useful as shared functionality, maybe something for `kibana-react`? quick question to @vadimkibana, isn't there a similar function in `react-use` we might have overlooked?